src/shogun/lib/any.h

```diff
@@ -655,7 +677,10 @@ namespace shogun {
 			{
 				return 0;
 			}
-			SG_FREE(ptr);
+			if constexpr (is_shared_ptr<T>::value)
+				ptr->reset();
```

Comment:
call the shared_ptr reset method, instead of deleting with free. Not sure what happens with unique_ptr...

Reply:
nvm, the issue is that we should call `delete[]` because this (could) represent an array with multiple pointers

Reply:
and each pointer managed by shared_ptr manages another ptr, so basically have a 2D array.. I think that's it!

Reply:
so have to delete the shared_ptr pointer and call it's destructor to destroy the managed pointer.. so `delete[] ptr`;